Error: If you try to add a new compute node or upgrade existing compute node or reconfigure existing compute node
       and if you happen to install openstack-nova-compute package again then this error might occure.

       This error does not appear in nova-compute logs however it appears when we check the status of
       openstack-nova-compute service.

      This looks like below:


[root@compute1 ~]# systemctl status openstack-nova-compute.service -l
● openstack-nova-compute.service - OpenStack Nova Compute Server
   Loaded: loaded (/usr/lib/systemd/system/openstack-nova-compute.service; enabled; vendor preset: disabled)
   Active: active (running) since Tue 2016-07-05 13:08:23 IST; 5min ago
 Main PID: 2489 (nova-compute)
   CGroup: /system.slice/openstack-nova-compute.service
           └─2489 /usr/bin/python2 /usr/bin/nova-compute

Jul 05 13:08:23 compute1 nova-compute[2489]: File "/usr/lib/python2.7/site-packages/oslo_messaging/transport.py", line 90, in _send
Jul 05 13:08:23 compute1 nova-compute[2489]: timeout=timeout, retry=retry)
Jul 05 13:08:23 compute1 nova-compute[2489]: File "/usr/lib/python2.7/site-packages/oslo_messaging/_drivers/amqpdriver.py", line 431, in send
Jul 05 13:08:23 compute1 nova-compute[2489]: retry=retry)
Jul 05 13:08:23 compute1 nova-compute[2489]: File "/usr/lib/python2.7/site-packages/oslo_messaging/_drivers/amqpdriver.py", line 422, in _send
Jul 05 13:08:23 compute1 nova-compute[2489]: raise result
Jul 05 13:08:23 compute1 nova-compute[2489]: RemoteError: Remote error: OperationalError (_mysql_exceptions.OperationalError) (1241, 'Operand should contain 1 column(s)') [SQL: u'SELECT migrations.created_at AS migrations_created_at, migrations.updated_at AS migrations_updated_at, migrations.deleted_at AS migrations_deleted_at, migrations.deleted AS migrations_deleted, migrations.id AS migrations_id, migrations.source_compute AS migrations_source_compute, migrations.dest_compute AS migrations_dest_compute, migrations.source_node AS migrations_source_node, migrations.dest_node AS migrations_dest_node, migrations.dest_host AS migrations_dest_host, migrations.old_instance_type_id AS migrations_old_instance_type_id, migrations.new_instance_type_id AS migrations_new_instance_type_id, migrations.instance_uuid AS migrations_instance_uuid, migrations.status AS migrations_status, migrations.migration_type AS migrations_migration_type, migrations.hidden AS migrations_hidden \nFROM migrations \nWHERE migrations.deleted = %s AND migrations.status = %s AND migrations.source_compute = %s AND migrations.migration_type = %s'] [parameters: (0, [u'accepted', u'done'], 'compute1', 'evacuation')]
Jul 05 13:08:23 compute1 nova-compute[2489]: [u'Traceback (most recent call last):\n', u'  File "/usr/lib/python2.7/site-packages/nova/conductor/manager.py", line 447, in _object_dispatch\n    return getattr(target, method)(*args, **kwargs)\n', u'  File "/usr/lib/python2.7/site-packages/oslo_versionedobjects/base.py", line 171, in wrapper\n    result = fn(cls, context, *args, **kwargs)\n', u'  File "/usr/lib/python2.7/site-packages/nova/objects/migration.py", line 157, in get_by_filters\n    db_migrations = db.migration_get_all_by_filters(context, filters)\n', u'  File "/usr/lib/python2.7/site-packages/nova/db/api.py", line 484, in migration_get_all_by_filters\n    return IMPL.migration_get_all_by_filters(context, filters)\n', u'  File "/usr/lib/python2.7/site-packages/nova/db/sqlalchemy/api.py", line 4500, in migration_get_all_by_filters\n    return query.all()\n', u'  File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/query.py", line 2588, in all\n    return list(self)\n', u'  File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/query.py", line 2736, in __iter__\n    return self._execute_and_instances(context)\n', u'  File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/query.py", line 2751, in _execute_and_instances\n    result = conn.execute(querycontext.statement, self._params)\n', u'  File "/usr/lib64/python2.7/site-packages/sqlalchemy/engine/base.py", line 914, in execute\n    return meth(self, multiparams, params)\n', u'  File "/usr/lib64/python2.7/site-packages/sqlalchemy/sql/elements.py", line 323, in _execute_on_connection\n    return connection._execute_clauseelement(self, multiparams, params)\n', u'  File "/usr/lib64/python2.7/site-packages/sqlalchemy/engine/base.py", line 1010, in _execute_clauseelement\n    compiled_sql, distilled_params\n', u'  File "/usr/lib64/python2.7/site-packages/sqlalchemy/engine/base.py", line 1146, in _execute_context\n    context)\n', u'  File "/usr/lib64/python2.7/site-packages/sqlalchemy/engine/base.py", line 1337, in _handle_dbapi_exception\n    util.raise_from_cause(newraise, exc_info)\n', u'  File "/
Jul 05 13:08:23 compute1 nova-compute[2489]: usr/lib64/python2.7/site-packages/sqlalchemy/util/compat.py", line 200, in raise_from_cause\n    reraise(type(exception), exception, tb=exc_tb)\n', u'  File "/usr/lib64/python2.7/site-packages/sqlalchemy/engine/base.py", line 1139, in _execute_context\n    context)\n', u'  File "/usr/lib64/python2.7/site-packages/sqlalchemy/engine/default.py", line 450, in do_execute\n    cursor.execute(statement, parameters)\n', u'  File "/usr/lib64/python2.7/site-packages/MySQLdb/cursors.py", line 174, in execute\n    self.errorhandler(self, exc, value)\n', u'  File "/usr/lib64/python2.7/site-packages/MySQLdb/connections.py", line 36, in defaulterrorhandler\n    raise errorclass, errorvalue\n', u"OperationalError: (_mysql_exceptions.OperationalError) (1241, 'Operand should contain 1 column(s)') [SQL: u'SELECT migrations.created_at AS migrations_created_at, migrations.updated_at AS migrations_updated_at, migrations.deleted_at AS migrations_deleted_at, migrations.deleted AS migrations_deleted, migrations.id AS migrations_id, migrations.source_compute AS migrations_source_compute, migrations.dest_compute AS migrations_dest_compute, migrations.source_node AS migrations_source_node, migrations.dest_node AS migrations_dest_node, migrations.dest_host AS migrations_dest_host, migrations.old_instance_type_id AS migrations_old_instance_type_id, migrations.new_instance_type_id AS migrations_new_instance_type_id, migrations.instance_uuid AS migrations_instance_uuid, migrations.status AS migrations_status, migrations.migration_type AS migrations_migration_type, migrations.hidden AS migrations_hidden \\nFROM migrations \\nWHERE migrations.deleted = %s AND migrations.status = %s AND migrations.source_compute = %s AND migrations.migration_type = %s'] [parameters: (0, [u'accepted', u'done'], 'compute1', 'evacuation')]\n"].
Jul 05 13:08:23 compute1 nova-compute[2489]: Removing descriptor: 5



Solutions:

Either you upgrade the packages on controller node or downgrade the packages on newly added or reconfigured compute node.
